ui/src/pages/AccountCreation/CreateAccount.js

```diff
@@ -0,0 +1,246 @@
+  const location = useLocation();
+
+  const checkPasswords = () => {
+
+    if (!passwordsMatch || !isValidPassword) {
+      setShowError(true);
+    } else {
+      goForward();
+    }
+
+  }
+
+  useEffect(() => {
+
+    const checkPasswordValidity = () => {
+      if (isValid(password)) {
+        setIsValidPassword(true);
+      } else {
+        setIsValidPassword(false);
+      }
+    }
+  
+    const checkPasswordMatch = () => {
+      if (password === confirmPassword) {
+        setPasswordsMatch(true);
+      } else {
+        setPasswordsMatch(false);
+      }
+    }
+
+    checkPasswordMatch();
+    checkPasswordValidity();
+
+  }, [password, confirmPassword, isValidPassword, passwordsMatch]);
+
+  // determines whether or not the inputted recovery phrase matches the one
+  // originally provided.
+  useEffect(() => {
+
+    const isValidPhraseSelection = () => {
+      // converting both arrays to strings in order to compare them
+      let stringedActual = JSON.stringify(actualPhrase);
+      let selectedRecoveryPhraseCopy = [...selectedRecoveryPhrase];
+
+      // removing indexes from words to compare properly in case of duplicate words.
+      for (let i = 0; i < selectedRecoveryPhraseCopy.length; i++) {
+        selectedRecoveryPhraseCopy[i] = selectedRecoveryPhraseCopy[i].split("_")[0];
+      }
+      let stringedSelected = JSON.stringify(selectedRecoveryPhraseCopy);
+
+      if (selectedRecoveryPhrase.length !== recoveryPhrase.split(" ").length) return;
+      if ((!isValidSelectedPhrase) &&
+        (stringedSelected === stringedActual)) {
+        console.log("[INFO]: Valid phrase chosen.");
+        setIsValidSelectedPhrase(true);
+      } else if ((isValidSelectedPhrase) &&
+        (stringedSelected !== stringedActual)) {
+        console.log("[INFO]: Invalid phrase chosen.")
+        setIsValidSelectedPhrase(false);
+      }
+    }
+
+    isValidPhraseSelection();
+
+  }, [selectedRecoveryPhrase, actualPhrase, isValidSelectedPhrase, recoveryPhrase]);
+
+  useEffect(() => {
+
+    const shuffleRecoveryPhrase = () => {
+      // shuffles array so that user gets recovery phrase in a different order.
+      let actualRecoveryPhrase = recoveryPhrase.split(" ");
+
+      let shuffled = actualRecoveryPhrase
+        .map(value => ({ value, sort: Math.random() }))
+        .sort((a, b) => a.sort - b.sort)
+        .map(({ value }) => value);
+
+      for (let i = 0; i < shuffled.length; i++) {
+        shuffled[i] = shuffled[i] + "_" + i;
+      }
+
+      // adding the item index so that we can distinguish between duplicate words.
+
+      setShuffledRecoveryPhrase(shuffled);
+      setActualPhrase(actualRecoveryPhrase);
+    }
+
+    shuffleRecoveryPhrase();
+
+  }, [recoveryPhrase]);
+
+  // onClickSelectWordButton is called whenever a word selection button
+  // gets clicked when the user is reciting their seed phrase.
+  const onClickSelectWordButton = (e, word) => {
+
+    if (selectedRecoveryPhrase.includes(word)) {
```

Comment:
I think this breaks if you have the same word more than once

Reply:
<img width="272" alt="image" src="https://user-images.githubusercontent.com/53206800/200446393-411b7ec2-4b59-4692-bcbe-fbb1aabcd19f.png">
I did get weird behaviour when there was more than one of each word, which is why I add an index tag for each word so that they're all unique. word in this function has an index tag like in the picture above

Reply:
I added a comment to clear up the confusion